### PR TITLE
Re-enablement of bf16 jit:uni reorders for aarch64

### PR DESCRIPTION
**Pull request by Shreyas-fuj**
# Description

The PR: https://github.com/oneapi-src/oneDNN/pull/1986 changed a particular condition for bf_16(This was not intended in the PR). So after this was merged some testcases which were going to jin:uni before, started going to simple:any(reference). Therefore, this PR contains the fix for that by just reverting the un-intended change to the original.

# Checklist

## General

- [y] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
```
make test

99% tests passed, 2 tests failed out of 200

Total Test time (real) = 2142.22 sec

The following tests FAILED:
	159 - test_graph_unit_dnnl_large_partition_usm_cpu (Failed)
	181 - test_benchdnn_modeC_graph_ci_cpu (Failed)
Errors while running CTest
Output from these tests are in: /home/shreyas/G/shr-fuj/oneDNN_open_source/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make: *** [Makefile:71: test] Error 8
```

- [y] Have you formatted the code using clang-format?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #===============================================================================
 
 build
+build_acl
```
Comment (theComputeKid) [inline .gitignore:19]: This looks like a personalised build folder. What is the reason for adding this, and not having cmake nest build folders inside a parent build folder which is already ignored?

**Review by theComputeKid — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Shreyas-fuj) [reply .gitignore]: Hi @theComputeKid,
I have restored gitignore, thanks.

**Review by dzarukin — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (mgouicem) [reply .gitignore]: I think it should be safe to use `build*` in .gitignore if devs want to use multiple build folders prefixed with build.

**Comment by mgouicem** [comment]
Could you please use commit messages the are aligned with the [contributing guideline](https://github.com/oneapi-src/oneDNN/blob/main/CONTRIBUTING.md#code-contribution-guidelines).
For example something like `aarch64: reenable bf16 jit reorders`

File: .gitignore (hunk 1 above)
Comment (theComputeKid) [reply .gitignore]: Agree completely with @mgouicem

**Comment by theComputeKid** [comment]
@Radu2k  could you please review this for the `aarch64` team? Thanks.

**Comment by Radu2k** [comment]
> Could you please use commit messages the are aligned with the [contributing guideline](https://github.com/oneapi-src/oneDNN/blob/main/CONTRIBUTING.md#code-contribution-guidelines). For example something like `aarch64: reenable bf16 jit reorders`

I agree with Mourad on this topic and want to highly endorse the importance of PR message format as this will help speeding up the review process. By taking the time to **correctly** label the PR, we can easily identify and asses issues based on their commit message labels, therefore reviewing and closing your PRs sooner.
Here is the contribution guidance for your convenience: https://github.com/oneapi-src/oneDNN/blob/main/CONTRIBUTING.md#code-contribution-guidelines

**Review by Radu2k — CHANGES_REQUESTED** [review]

File: src/cpu/aarch64/jit_uni_reorder.cpp [hunk 2]
```diff
@@ -1285,17 +1291,17 @@ struct jit_uni_reorder_kernel_f32_t : public kernel_t, public jit_generator {
         }
     }
 
-    bool interim_f32_needed() {
+    static bool interim_f32_needed(const prb_t &prb, bool compensation_needed) {
```
[the remark this answers is not in the document]
Comment (Shreyas-fuj) [reply src/cpu/aarch64/jit_uni_reorder.cpp:1294]: Hi @Radu2k , as mentioned this is just reversal of an unintended change in out PR. We have not done any additional changes ourselves. This was present before our PR was merged, so we have just restored it as is.

**Comment by Shreyas-fuj** [comment]
@mgouicem 
I have squashed, renamed and pushed the commit, thanks.

**Review by Radu2k — APPROVED** [review]
LGTM.

**Review by mgouicem — APPROVED** [review]

**Comment by theComputeKid** [comment]
Going to check perf on g3 and then will approve.

**Comment by Shreyas-fuj** [comment]
> Going to check perf on g3 and then will approve.

Hi @theComputeKid , any update on this? Thanks.

**Review by theComputeKid — APPROVED** [review]

**Pull request merged**